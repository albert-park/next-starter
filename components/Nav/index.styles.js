import css from 'styled-jsx/css';
import styled from 'styled-components';

export default css`
  :global(body) {
    margin: 0;
    font-family: -apple-system, BlinkMacSystemFont, Avenir Next, Avenir,
      Helvetica, sans-serif;
  }
`;

export const StyledNav = styled.nav`
  text-align: center;
`;

export const StyledUl = styled.ul`
    display: flex;
    justify-content: space-between;
    padding: 4px 16px;
`;

export const StyledLi = styled.li`
    display: flex;
    padding: 6px 8px;
`;

export const StyledAnchor = styled.a`
    color: #067df7;
    text-decoration: none;
    font-size: 13px;
`;
