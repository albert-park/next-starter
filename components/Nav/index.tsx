import React, { FC } from 'react';
import Link from 'next/link';
import NavStyles from './index.styles';

export const links = [
  { href: '/', label: 'Home', key: '' },
  { href: 'https://zeit.co/now', label: 'ZEIT', key: '' },
  { href: 'https://github.com/zeit/next.js', label: 'GitHub', key: '' },
].map((link) => {
  Object.assign(link, { key: `nav-link-${link.href}-${link.label}` });
  return link;
});

const Nav: FC = () => (
  <nav>
    <ul>
      {links.map(({ key, href, label }) => (
        <li key={key}>
          <a href={href}>{label}</a>
        </li>
      ))}
    </ul>

    <style jsx>
      {NavStyles}
    </style>
  </nav>
);

export default Nav;
