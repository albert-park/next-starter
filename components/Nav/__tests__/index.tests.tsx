import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Nav, { links } from '../index';

const renderTest = () => {
  return (render(<Nav />))
} 

describe('Nav Component', () => {
  it('should match snapshot', () => {
    const { asFragment } = renderTest();
    expect(asFragment()).toMatchSnapshot();
  });
  links.map((link) => (
    it('renders each link text correctly', () => {
      const { getByText } = renderTest();
      expect(getByText(link.label)).toHaveTextContent(link.label);
      expect(getByText(link.label)).toBeVisible();
    })));

  it('renders the correct number of links', () => {
    const { container } = renderTest();
    expect(container.querySelectorAll('li')).toHaveLength(links.length);
  });
});
