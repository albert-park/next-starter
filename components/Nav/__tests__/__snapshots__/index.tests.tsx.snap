// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav Component should match snapshot 1`] = `
<DocumentFragment>
  <nav
    class="jsx-3319782719"
  >
    <ul
      class="jsx-3319782719"
    >
      <li
        class="jsx-3319782719"
      >
        <a
          class="jsx-3319782719"
          href="/"
        >
          Home
        </a>
      </li>
      <li
        class="jsx-3319782719"
      >
        <a
          class="jsx-3319782719"
          href="https://zeit.co/now"
        >
          ZEIT
        </a>
      </li>
      <li
        class="jsx-3319782719"
      >
        <a
          class="jsx-3319782719"
          href="https://github.com/zeit/next.js"
        >
          GitHub
        </a>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;
